### ISSUE TITLE
Show correct line number in error

### ISSUE THREAD

**Issue by arnoson** [blackketter/LuaArduino#4]
Right now every error has the line number 0 (when executing a string or a file). Is there any way to show the actual line number?

**Comment by blackketter** [comment]
Could you give an example with some code?  Thanks.

**Comment by arnoson** [comment]
If I use a basic setup and then load a string like this:
```c++
  lua->loadString(
    "local something = 10\n"
    "thisDoesntExist()\n"
  );
```
I get the error message
```
[string "local something = 10..."]:0: attempt to call global 'thisDoesntExist' (a nil value)
```
But I'd expect the following, as `0` is not a valid line number.
```
[string "local something = 10..."]:2: attempt to call global 'thisDoesntExist' (a nil value)
```

**Comment by arnoson** [comment]
It seems to be related to the debug optimization. If I disable `LUA_OPTIMIZE_DEBUG` in `luaconf.h` the line numbers work again.
```c++
#define LUA_OPTIMIZE_DEBUG 0
```
Not sure however how much this will effect memory usage or performance.

**Comment by arnoson** [comment]
I did some more tests and it seems I was wrong in stating that the line number is missing in every error. It was only the case in the errors I got most recently. The errors that are missing the line number involve either calling a nil function like `doesNotExist()` or calling a function of a nil object like `doesNotExist.myFunction()`. I guess as soon as nil functions are involved the scope and therefore the line number gets lost. For now I will keep the debug optimization disabled until I run into performance issues.
